opm/parser/eclipse/EclipseState/EndpointScaling.cpp

```diff
@@ -0,0 +1,113 @@
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/EclipseState/Runspec.hpp>
+#include <opm/parser/eclipse/Utility/String.hpp>
+
+namespace Opm {
+
+EndpointScaling::operator bool() const noexcept {
+    return this->options[ static_cast< ue >( option::any ) ];
+}
+
+bool EndpointScaling::directional() const noexcept {
+    return this->options[ static_cast< ue >( option::directional ) ];
+}
+
+bool EndpointScaling::nondirectional() const noexcept {
+    return bool( *this ) && !this->directional();
+}
+
+bool EndpointScaling::reversible() const noexcept {
+    return this->options[ static_cast< ue >( option::reversible ) ];
+}
+
+bool EndpointScaling::irreversible() const noexcept {
+    return bool( *this ) && !this->reversible();
+}
+
+bool EndpointScaling::twopoint() const noexcept {
+    return bool( *this ) && !this->threepoint();
+}
+
+bool EndpointScaling::threepoint() const noexcept {
+    return this->options[ static_cast< ue >( option::threepoint ) ];
+}
+
+namespace {
+
+bool threepoint_scaling( const Deck& deck ) {
```

Comment:
Thank you for asking - I think this slightly tricky: See #1021 for my view of the situation.